build-support/bin/packages.py

```diff
@@ -231,11 +343,50 @@ def build_pants_wheels() -> None:
                     f"expecting only one wheel: {sorted(wheel.name for wheel in found_wheels)}."
                 )
             for wheel in found_wheels:
-                if not (destination / wheel.name).exists():
+                if not (dest / wheel.name).exists():
                     # We use `copy2` to preserve metadata.
-                    shutil.copy2(wheel, destination)
+                    shutil.copy2(wheel, dest)
 
-    green(f"Wrote Pants wheels to {destination}.")
+    green(f"Wrote Pants wheels to {dest}.")
+
+    banner(f"Validating Pants wheels for {CONSTANTS.python_version}.")
+    create_twine_venv()
+    subprocess.run([CONSTANTS.twine_venv_dir / "bin/twine", "check", dest / "*.whl"], check=True)
+    green(f"Validated Pants wheels for {CONSTANTS.python_version}.")
+
+
+def build_3rdparty_wheels() -> None:
+    banner(f"Building 3rdparty wheels with Python {CONSTANTS.python_version}")
+    dest = CONSTANTS.deploy_3rdparty_wheel_dir / CONSTANTS.pants_unstable_version
+    pkg_tgts = [pkg.target for pkg in PACKAGES]
+    with create_tmp_venv() as venv_tmpdir:
+        deps = (
+            subprocess.run(
+                [
+                    "./pants",
+                    "--concurrent",
+                    "--no-dynamic-ui",
+                    "dependencies",
+                    "--transitive",
+                    "--type=3rdparty",
+                    *pkg_tgts,
+                ],
+                stdout=subprocess.PIPE,
+                check=True,
+            )
+            .stdout.decode()
+            .strip()
+            .splitlines()
+        )
+        if not deps:
+            raise AssertionError(
+                f"No 3rd-party dependencies detected for {pkg_tgts}. Is `./pants dependencies` "
+                "broken?"
+            )
```

Comment:
This is the main change from Bash, which used to read our requirements file instead.